lib/rubocop/cop/style/method_call_with_args_parentheses.rb

```diff
@@ -212,53 +210,85 @@ def args_parenthesized?(node)
           node.super_type? && node.arguments.none?
         end
 
+        def camel_case_method_call_without_arguments?(node)
+          node.camel_case_method? && node.arguments.none?
+        end
+
         def eligible_for_parentheses_presence?(node)
-          node.implicit_call? ||
-            call_in_arguments_or_literals?(node) ||
+          call_in_literals?(node) ||
             call_with_ambiguous_arguments?(node) ||
             call_in_logical_operators?(node) ||
             allowed_multiline_call_with_parentheses?(node) ||
             allowed_chained_call_with_parentheses?(node)
         end
 
-        def call_in_arguments_or_literals?(node)
+        def call_in_literals?(node)
           node.parent &&
-            (node.parent.send_type? ||
-             node.parent.pair_type? ||
-             node.parent.array_type?)
+            (node.parent.pair_type? ||
+             node.parent.array_type? ||
+             ternary_if?(node.parent))
+        end
+
+        def call_in_logical_operators?(node)
+          node.parent &&
+            (logical_operator?(node.parent) ||
+             node.parent.descendants.any?(&method(:logical_operator?)))
         end
 
         def call_with_ambiguous_arguments?(node)
-          node.block_node && node.block_node.braces? ||
+          call_with_braced_block?(node) ||
+            hash_literal_in_arguments?(node) ||
             node.descendants.any? do |n|
-              n.splat_type? || n.kwsplat_type? || n.block_pass_type?
+              splat?(n) || ternary_if?(n) || logical_operator?(n)
             end
         end
 
-        def call_in_logical_operators?(node)
-          node.descendants.any?(&LOGICAL_OPERATOR_CHECK) || (node.parent &&
-            (LOGICAL_OPERATOR_CHECK.call(node.parent) ||
-             node.parent.descendants.any?(&LOGICAL_OPERATOR_CHECK)))
+        def call_with_braced_block?(node)
+          node.block_node && node.block_node.braces?
+        end
+
+        def hash_literal_in_arguments?(node)
+          node.arguments.any? do |n|
+            hash_literal?(n) ||
+              n.send_type? && node.descendants.any?(&method(:hash_literal?))
+          end
         end
 
         def allowed_multiline_call_with_parentheses?(node)
           cop_config['AllowParenthesesInMultilineCall'] && node.multiline?
         end
 
         def allowed_chained_call_with_parentheses?(node)
-          cop_config['AllowParenthesesInChaining'] &&
-            node.descendants.first && node.descendants.first.send_type?
+          (node.parent && node.parent.send_type?) ||
+            cop_config['AllowParenthesesInChaining'] &&
+              node.descendants.first && node.descendants.first.send_type?
         end
 
-        def parentheses_at_the_end_of_multiline_call?(node)
-          node.multiline? &&
-            node.loc.begin.source_line
-                .gsub(TRAILING_WHITESPACE_REGEX, '')
-                .end_with?('(')
+        def splat?(node)
```

Comment:
Some of the methods here exist to reduce the cyclomatic complexity of the bigger ones. Since the parentheses omissions are quite context sensitive, I can't run away from all those conditions. Is it okay to special-comment the offending methods?